substratools/algo.py

```diff
@@ -99,18 +100,16 @@ def execute(self, method_name: str, rank: int = 0, fake_data: bool = False, n_fa
         # load inputs
         inputs = deepcopy(self._workspace.task_inputs)
 
-        inputs.update({"rank": rank})
+        task_properties = {"rank": rank}
```

Comment:
out of scope, but "rank" should be a variable like `TASK_IO_DATASAMPLES`
and the following variables should be in an Enum:

```python
TASK_IO_OPENER = "opener"
TASK_IO_DATASAMPLES = "datasamples"
TASK_IO_CHAINKEYS = "chainkeys"
```

Reply:
done